src/devices/Vl53L1X/Vl53L1X.cs

```diff
@@ -0,0 +1,1022 @@
+            0x68, /* 0x65 : Sigma threshold LSB */
+            0x00, /* 0x66 : Min count Rate MSB (MCPS in 9.7 format for MSB+LSB), use SetSignalThreshold() */
+            0x80, /* 0x67 : Min count Rate LSB */ 0x08, /* 0x68 : not user-modifiable */
+            0xb8, /* 0x69 : not user-modifiable */ 0x00, /* 0x6a : not user-modifiable */
+            0x00, /* 0x6b : not user-modifiable */
+            0x00, /* 0x6c : Intermeasurement period MSB, 32 bits register, use SetIntermeasurementInMs() */
+            0x00, /* 0x6d : Intermeasurement period */ 0x0f, /* 0x6e : Intermeasurement period */
+            0x89, /* 0x6f : Intermeasurement period LSB */ 0x00, /* 0x70 : not user-modifiable */
+            0x00, /* 0x71 : not user-modifiable */
+            0x00, /* 0x72 : distance threshold high MSB (in mm, MSB+LSB), use SetD:tanceThreshold() */
+            0x00, /* 0x73 : distance threshold high LSB */
+            0x00, /* 0x74 : distance threshold low MSB ( in mm, MSB+LSB), use SetD:tanceThreshold() */
+            0x00, /* 0x75 : distance threshold low LSB */ 0x00, /* 0x76 : not user-modifiable */
+            0x01, /* 0x77 : not user-modifiable */ 0x0f, /* 0x78 : not user-modifiable */
+            0x0d, /* 0x79 : not user-modifiable */ 0x0e, /* 0x7a : not user-modifiable */
+            0x0e, /* 0x7b : not user-modifiable */ 0x00, /* 0x7c : not user-modifiable */
+            0x00, /* 0x7d : not user-modifiable */ 0x02, /* 0x7e : not user-modifiable */
+            0xc7, /* 0x7f : ROI center, use SetROI() */ 0xff, /* 0x80 : XY ROI (X=Width, Y=Height), use SetROI() */
+            0x9B, /* 0x81 : not user-modifiable */ 0x00, /* 0x82 : not user-modifiable */
+            0x00, /* 0x83 : not user-modifiable */ 0x00, /* 0x84 : not user-modifiable */
+            0x01, /* 0x85 : not user-modifiable */ 0x00, /* 0x86 : clear interrupt, use ClearInterrupt() */
+            0x00 /* 0x87 : start ranging, use StartRanging() or StopRanging(), If you want an automatic start after VL53L1X_init() call, put 0x40 in location 0x87 */
+        };
+
+        private I2cDevice _i2CDevice = null!;
+        private bool _rangingInitialized;
+        private GpioController? _controller;
+
+        /// <summary>
+        /// Creates a Vl53L1X sensor class
+        /// </summary>
+        /// <param name="i2cAddress">The I2C address for the device</param>
+        /// <param name="busId">The id of the I2C bus</param>
+        /// <param name="xShutPin">
+        /// The xShut pin number used to power up the device.
+        /// If omitted, The xShutPin must be tied to the power supply value through a pull up resistor
+        /// </param>
+        /// <param name="operationTimeoutMilliseconds">Timeout for reading data, by default 500 milliseconds</param>
+        public Vl53L1X(byte i2cAddress = (byte)Registers.VL53L1X_DEFAULT_DEVICE_ADDRESS, int busId = 1, int xShutPin = -1, int operationTimeoutMilliseconds = 500)
```

Comment:
This is not allowed and we only pass the I2cDevice. See https://github.com/dotnet/iot/blob/b1cee19c6223a8755bd6fabff41f2f6332ced73c/src/devices/Vl53L0X/Vl53L0X.cs#L69 for a similar device.

Reply:
I'm now passing the device over the constructor